src/EnergyPlus/UnitarySystem.cc

```diff
@@ -15090,27 +15063,19 @@ namespace UnitarySystems {
                                                                 PartLoadFrac); // QCoilReq, simulate any load > 0 to get max capacity
                     } break;
                     case DataHVACGlobals::Coil_UserDefined: { // do nothing, user defined coil cannot be controlled
-                        UserDefinedComponents::SimCoilUserDefined(state, CompName, CompIndex, AirLoopNum, HeatingActive, CoolingActive);
+                        UserDefinedComponents::SimCoilUserDefined(
+                            state, CompName, this->m_SuppHeatCoilIndex, AirLoopNum, HeatingActive, CoolingActive);
                         if (HeatingActive) PartLoadFrac = 1.0;
                     } break;
                     default:
                         break;
                     }
 
-                    NoLoadTempOut = state.dataLoopNodes->Node(OutletNode).Temp;
-                    //      NoOutput = state.dataLoopNodes->Node(InletNode)%MassFlowRate *  &
-                    //                       (PsyHFnTdbW(NoLoadTempOut,Node(OutletNode)%HumRat)  &
-                    //                        - PsyHFnTdbW(Node(InletNode)%Temp,Node(OutletNode)%HumRat))
-
-                    //     If OutletTemp is within ACC of set point, either coil operated or is not needed
-                    if (std::abs(state.dataLoopNodes->Node(OutletNode).Temp - DesOutTemp) < Acc ||
-                        this->m_SuppHeatCoilType_Num == DataHVACGlobals::Coil_UserDefined) {
-                        // do nothing, coil is at set point (i.e., gas/elec/steam/user coil will try to hit set point)
-                    } else if (PartLoadFrac > 0.0) {
-                        // do nothing, coil tried to hit set point (i.e., gas/elec/steam/user coil tried to hit set point but missed
-                    } else if (NoLoadTempOut > (DesOutTemp - Acc)) {
-                        PartLoadFrac = 0.0; // outlet temp > set point, coil is not needed
-                    } else {                // outlet temp too low, turn on coil
+                    // If OutletTemp is within ACC of set point coil operation is not needed or UserDefined already met load.
+                    if (outletNode.Temp > (DesOutTemp - Acc) || this->m_SuppHeatCoilType_Num == DataHVACGlobals::Coil_UserDefined) {
+                        // do nothing, coil is at or above set point
+                    } else {
+                        // outlet temp too low, turn on coil
```

Comment:
More clean up. PartLoadFrac = 0 here so that conditional was never true. Double test of outlet temp also reduced to single conditional.